src/app/components/messages/messages.component.ts

```diff
@@ -50,40 +39,33 @@ export interface SaveEvent {
   styleUrl: './messages.component.scss',
   animations: slideInOut,
 })
-export class MessagesComponent implements OnInit, OnDestroy, AfterViewInit {
-  @Input() filteredMessages!: Message[]
-  @Input() scroll!: CdkVirtualScrollViewport
-  @Input() animationState!: string
+export class MessagesComponent implements OnInit, AfterViewInit {
+  readonly animationState = input.required<AnimationState>()
 
-  @ViewChildren('pre') messageElements!: QueryList<ElementRef>
+  readonly filteredMessages = input.required<Message[]>()
 
-  @Output() save = new EventEmitter<SaveEvent>()
-  @Output() dataEmitted = new EventEmitter<number>()
+  readonly messageElements: Signal<readonly ElementRef<HTMLElement>[]> = viewChildren('pre')
```

Comment:
Is it possible to specify type when calling `viewChildren`?

E.g.
```suggestion
  readonly messageElements = viewChildren<ElementRef<HTMLElement>[]>('pre')
```